### PR TITLE
Add starboardized CPU detection to WebP

### PR DESCRIPTION
**Pull request by kaidokert**
b/290060548

### CODE REVIEW

File: third_party/libwebp/src/dsp/cpu.c [hunk 1]
```diff
@@ -13,6 +13,51 @@
 
 #include "src/dsp/dsp.h"
 
+#if defined(STARBOARD)
+
+#include "starboard/cpu_features.h"
+#include "starboard/log.h"
+#include "starboard/system.h"
+
+bool warn_if_not_enabled(const char * name, bool value) {
```
Comment (hlwarriner) [inline third_party/libwebp/src/dsp/cpu.c:22]: nit: when referring to a pointer I think there should be a space either before or after the asterisk, but not both - go/cstyle#Pointer_and_Reference_Expressions.

File: third_party/libwebp/src/dsp/cpu.c [hunk 2]
```diff
@@ -13,6 +13,51 @@
...
+  if(!value) {
```
Comment (hlwarriner) [inline third_party/libwebp/src/dsp/cpu.c:23]: nit: should be a space here.

**Review by hlwarriner — CHANGES_REQUESTED** [review]

File: third_party/libwebp/BUILD.gn [hunk 3]
```diff
@@ -87,6 +90,27 @@ libwebp_lib("libwebp_utils_enc") {
 
 config("libwebp_direct_config") {
   include_dirs = [ "." ]
+  defines = []
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply third_party/libwebp/BUILD.gn]: Done - i also had an unpushed change that enables SSE on x86/Atom that further changes this bit. Please have another look

File: third_party/libwebp/src/dsp/cpu.c (hunk 1 above)
Comment (kaidokert) [reply third_party/libwebp/src/dsp/cpu.c]: Well what do you know, i found that we don't run clang-format by default on C files at all, even if i remove third_party excludes. Just did that and it cleaned up rest of spacing below as well.

File: third_party/libwebp/src/dsp/cpu.c (hunk 2 above)
Comment (kaidokert) [reply third_party/libwebp/src/dsp/cpu.c]: Done along with rest of clang-format

**Pull request merged**